File: packages/api-client/src/appointments.ts
import { apiClient } from './client';
import type { Appointment, CreateAppointmentRequest, AppointmentSlot } from '@stratosphere/types';

export const appointmentsApi = {
  getAppointments: async (params?: { status?: string; limit?: number }): Promise<{ 
    appointments: Appointment[]; 
    total: number 
  }> => {
    const response = await apiClient.get('/appointments', { params });
    return response.data;
  },

  createAppointment: async (data: CreateAppointmentRequest): Promise<Appointment> => {
    const response = await apiClient.post('/appointments', data);
    return response.data;
  },

  getAppointment: async (id: string): Promise<Appointment> => {
    const response = await apiClient.get(`/appointments/${id}`);
    return response.data;
  },

  updateAppointment: async (id: string, data: { scheduledAt?: string }): Promise<Appointment> => {
    const response = await apiClient.put(`/appointments/${id}`, data);
    return response.data;
  },

  cancelAppointment: async (id: string): Promise<void> => {
    await apiClient.delete(`/appointments/${id}`);
  },

  getAvailableSlots: async (params: { 
    startDate: string; 
    endDate?: string; 
    appointmentType?: string 
  }): Promise<{ slots: AppointmentSlot[] }> => {
    const response = await apiClient.get('/appointments/available-slots', { params });
    return response.data;
  },
};

File: backend/app/api/v1/appointments.py
from fastapi import APIRouter, Query
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime, date

router = APIRouter()


class AppointmentCreate(BaseModel):
    appointment_type: str  # routine, follow_up, urgent, phone, video
    reason: str | None = None
    preferred_date: date | None = None
    preferred_time: str | None = None  # morning, afternoon, evening


class AppointmentResponse(BaseModel):
    id: UUID
    appointment_type: str
    status: str
    scheduled_at: datetime | None
    duration_minutes: int
    created_at: datetime


@router.get("/")
async def list_appointments(
    status: str | None = None,
    limit: int = Query(default=20, le=100),
):
    """List appointments"""
    return {"appointments": [], "total": 0}


@router.post("/", response_model=AppointmentResponse)
async def request_appointment(request: AppointmentCreate):
    """Request a new appointment"""
    return AppointmentResponse(
        id=UUID("00000000-0000-0000-0000-000000000000"),
        appointment_type=request.appointment_type,
        status="requested",
        scheduled_at=None,
        duration_minutes=15,
        created_at=datetime.utcnow(),
    )


@router.get("/available-slots")
async def get_available_slots(
    start_date: date,
    end_date: date | None = None,
    appointment_type: str = "routine",
):
    """Get available appointment slots"""
    return {"slots": []}


@router.get("/{appointment_id}")
async def get_appointment(appointment_id: UUID):
    """Get appointment details"""
    return {"id": appointment_id}


@router.put("/{appointment_id}")
async def update_appointment(appointment_id: UUID, scheduled_at: datetime | None = None):
    """Update/reschedule appointment"""
    return {"id": appointment_id, "message": "Appointment updated"}


@router.delete("/{appointment_id}")
async def cancel_appointment(appointment_id: UUID):
    """Cancel appointment"""
    return {"message": "Appointment cancelled"}
